test/unit/core.js

```diff
@@ -439,6 +439,13 @@ QUnit.test( "isFunction", function( assert ) {
 	fn = function() {};
 	assert.ok( jQuery.isFunction( fn ), "Normal Function" );
 
+	// Make sure ES7 async functions also work.
+	// Because our code should work in ES5 we can't define `var asyncFn = async function() {};` directly.
+	function AsyncFunction() { }
```

Comment:
I'd rather test a real async function; this test more or less repeats the implementation so it's not a full test to me.

Also, this should be a separate `QUnit.test`.

So that this test doesn't crash in old browsers, `eval` should be used to define an async function and the test should only be run if async functions are supported. Other browsers should have `QUnit.skip` called, see [this core test](https://github.com/jquery/jquery/blob/ecd8ddea33dc40ae2a57e4340be03faf2ba2f99b/test/unit/core.js#L383-L388) as an example.